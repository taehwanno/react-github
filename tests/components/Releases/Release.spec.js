import React from 'react';

import ReactGithub, { Release } from '../../../src';

describe('<Release />', () => {
  it('should throw an error when props.tagName not exist or not string', () => {
    expect(() => {
      ReactGithub.render(<Release />);
    }).toThrowErrorMatchingSnapshot();
  });

  it('should match snapshot', () => {
    expect(ReactGithub.render(<Release tagName="v1.0.0" />)).toMatchSnapshot();
  });

  it('should have target_commitish with props.targetCommitish', () => {
    const targetCommitish = 'production';
    const result = ReactGithub.render(<Release tagName="v1.0.0" targetCommitish={targetCommitish} />);
    expect(result.target_commitish).toBe(targetCommitish);
  });

  it('should have name with props.name', () => {
    const name = 'Major release';
    const result = ReactGithub.render(<Release tagName="v1.0.0" name={name} />);
    expect(result.name).toBe(name);
  });

  it('should have result with props.children', () => {
    const children = 'The first release';
    const result = ReactGithub.render(<Release tagName="v1.0.0">{children}</Release>);
    expect(result.body).toBe(children);
  });

  it('should have draft with props.draft', () => {
    const draft = true;
    const result = ReactGithub.render(<Release tagName="v1.0.0" draft={draft} />);
    expect(result.draft).toBe(draft);
  });

  it('should have prerelease with props.prerelease', () => {
    const prerelease = true;
    const result = ReactGithub.render(<Release tagName="v1.0.0" prerelease={prerelease} />);
    expect(result.prerelease).toBe(prerelease);
  });
});
