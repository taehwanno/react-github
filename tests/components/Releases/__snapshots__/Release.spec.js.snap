// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Release /> should match snapshot 1`] = `
Object {
  "body": "",
  "draft": false,
  "name": "v1.0.0",
  "prerelease": false,
  "tag_name": "v1.0.0",
  "target_commitish": "master",
}
`;

exports[`<Release /> should throw an error when props.tagName not exist or not string 1`] = `"props.tagName must have with string type"`;
